deisctl/backend/fleet/unit.go

```diff
@@ -63,8 +56,13 @@ func (c *FleetClient) lastUnit(component string) (num int, err error) {
 
 // NewUnit takes a component type and returns a Fleet unit
 // that includes the relevant systemd service template
-func NewUnit(component string) (uf *unit.UnitFile, err error) {
-	template, err := readTemplate(component)
+func NewUnit(c *FleetClient, component string) (uf *unit.UnitFile, err error) {
+	uf, err = doNewUnit(component, c.templatePaths)
```

Comment:
If NewUnit requires the client's templatePaths field, I feel like we should be making this a FleetClient function rather than a package function. That way we don't have to do this roundabout `doNewUnit` function. Any thoughts on this?